src/main/java/org/opensearch/plugin/insights/rules/model/SearchQueryRecord.java

```diff
@@ -130,7 +155,11 @@ public XContentBuilder toXContent(final XContentBuilder builder, final ToXConten
     @Override
     public void writeTo(final StreamOutput out) throws IOException {
         out.writeLong(timestamp);
-        out.writeMap(measurements, (stream, metricType) -> MetricType.writeTo(out, metricType), StreamOutput::writeGenericValue);
+        out.writeMap(
+            measurements,
+            (stream, metricType) -> MetricType.writeTo(out, metricType),
+            (stream, measurement) -> measurement.writeTo(out)
+        );
```

Comment:
This will break during the version between old and new nodes? I am wondering if we even need to implement the `Writeable` interface for `SearchQueryRecord`?

Reply:
This is used here : https://github.com/opensearch-project/query-insights/blob/main/src/main/java/org/opensearch/plugin/insights/rules/action/top_queries/TopQueries.java#L36

Reply:
Made the backward compatibility changes as discussed.